src/shapes.ts

```diff
@@ -0,0 +1,367 @@
+/**
+ * @packageDocumentation
+ * @module settings
+ *
+ * This file consists of shapes supported by chemiscope via 3dmol
+ *
+ * To add a new shape type, extend the `Shape` class and define the
+ * inherited functions (`validateParams` and `outputTo3Dmol`).
+ * `validateParams` should check a `ShapeData` object to see that
+ * it has the correct parameters for the given type. `outputTo3Dmol`
+ * should take a color and output a `CustomShapeSpec` (https://3dmol.csb.pitt.edu/doc/types.html#CustomShapeSpec)
+ * object. You will also need to add any parameters for your shape
+ * to ShapeData.
+ *
+ * To add a new shape, you will also need to add code to `structure/viewer.ts`.
+ */
+
+import { default as $3Dmol } from '3dmol';
+import { Quaternion } from '3dmol';
+
+// Just an interface to enforce XYZ-type coordinates
+export interface XYZ {
+    x: number;
+    y: number;
+    z: number;
+}
+export interface SphereData {
+    radius: number;
+    kind: string;
+}
+
+// Interface for ellipsoidal data, where
+// orientation is stored in the (w, x, y, z) convention
+export interface EllipsoidData {
+    semiaxes: [number, number, number];
+    orientation?: [number, number, number, number];
+    kind: string;
+}
+
+// Interface for polytope data, where
+// orientation is stored in the (w, x, y, z) convention
+export interface CustomShapeData {
+    vertices: [number, number, number][];
+    simplices: [number, number, number][];
+    orientation?: [number, number, number, number];
+    kind: string;
+}
+
+export type ShapeData = SphereData | EllipsoidData | CustomShapeData;
+
+export class Shape {
+    public position: [number, number, number];
+    /// orientation of the particle
```

Comment:
Could you add the `xyzw` convention here, and the fact that this is a quaternion?